ReportingServicesTools/Functions/CatalogItems/Rest/Out-RsRestFolderContent.ps1

```diff
@@ -0,0 +1,185 @@
+    }
+    Process
+    {
+        if ($RestApiVersion -eq 'v1.0')
+        {
+            try
+            {
+                Write-Verbose "Fetching $RsFolder info from server..."
+                $url = [string]::Format($catalogItemsByPathApiV1, $RsFolder)
+                if ($Credential -ne $null)
+                {
+                    $response = Invoke-WebRequest -Uri $url -Method Get -Credential $Credential -Verbose:$false
+                }
+                else
+                {
+                    $response = Invoke-WebRequest -Uri $url -Method Get -UseDefaultCredentials -Verbose:$false
+                }
+            }
+            catch
+            {
+                throw (New-Object System.Exception("Error while trying to fetch $RsFolder info! Exception: $($_.Exception.Message)", $_.Exception))
+            }
+
+            $folder = ConvertFrom-Json $response.Content
+
+            try
+            {
+                Write-Verbose "Fetching catalog items under $RsFolder from server..."
+                $url = [string]::Format($folderCatalogItemsApiV1, $folder.Id)
+                if ($Credential -ne $null)
+                {
+                    $response = Invoke-WebRequest -Uri $url -Method Get -Credential $Credential -Verbose:$false
+                }
+                else
+                {
+                    $response = Invoke-WebRequest -Uri $url -Method Get -UseDefaultCredentials -Verbose:$false
+                }
+            }
+            catch
+            {
+                throw (New-Object System.Exception("Error while trying to fetch catalog items under $RsFolder! Exception: $($_.Exception.Message)", $_.Exception))
+            }
+        }
+        else
+        {
+            try
+            {
+                Write-Verbose "Fetching catalog items under $RsFolder from server..."
+                $url = [string]::Format($folderCatalogItemsApiLatest, $RsFolder)
+                if ($Credential -ne $null)
+                {
+                    $response = Invoke-WebRequest -Uri $url -Method Get -Credential $Credential -Verbose:$false
+                }
+                else
+                {
+                    $response = Invoke-WebRequest -Uri $url -Method Get -UseDefaultCredentials -Verbose:$false
+                }
+            }
+            catch
+            {
+                throw (New-Object System.Exception("Error while trying to fetch catalog items under $RsFolder! Exception: $($_.Exception.Message)", $_.Exception))
+            }
+        }
+
+        $catalogItems = (ConvertFrom-Json $response.Content).value
+        foreach ($catalogItem in $catalogItems)
+        {
+            if ($catalogItem.Type -eq "Report" -or
+                $catalogItem.Type -eq "DataSet" -or
+                $catalogItem.Type -eq "DataSource" -or
+                $catalogItem.Type -eq "MobileReport" -or
+                $catalogItem.Type -eq "PowerBIReport")
```

Comment:
not exporting resources?

Reply:
Added support for Resources. We didn't have support for it in the existing Out-RsFolderContent, Out-RsCatalogItem, Write-RsFolderContent and Write-RsCatalogItem.

Reply:
Also added support for Excel Workbook.